mlrun/feature_store/retrieval/base.py

```diff
@@ -54,8 +71,10 @@ def start(
         with_indexes=None,
         update_stats=None,
         query=None,
+        join_type="inner",
     ):
         self._target = target
+        self._join_type = join_type
```

Comment:
This would be an amazing place to explode if the engine is `spark` and the `_join_type` is not `inner`. User should be aware that this is not supported.